From 9a4523d47efeafd674d419169fe161e5a3b31cb3 Mon Sep 17 00:00:00 2001
From: Jan Beich <jbeich@FreeBSD.org>
Date: Thu, 3 Jun 2021 17:53:11 +0000
Subject: [PATCH 1/3] view: chase swaywm/wlroots@9e58301df7f0

view.c:238:52: error: no member named 'subsurfaces' in 'struct wlr_surface'
        wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces, parent_link) {
                                       ~~~~~~~~~~~~~~~~~  ^
/usr/include/wayland-util.h:443:30: note: expanded from macro 'wl_list_for_each'
        for (pos = wl_container_of((head)->next, pos, member);  \
                                    ^~~~
/usr/include/wayland-util.h:409:32: note: expanded from macro 'wl_container_of'
        (__typeof__(sample))((char *)(ptr) -                            \
                                      ^~~

Based on https://github.com/swaywm/sway/commit/3162766eef14
---
 view.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/view.c b/view.c
index b9ba9c2..3f3b0ed 100644
--- a/view.c
+++ b/view.c
@@ -235,7 +235,10 @@ view_map(struct cg_view *view, struct wlr_surface *surface)
 	view->wlr_surface = surface;
 
 	struct wlr_subsurface *subsurface;
-	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces, parent_link) {
+	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces_below, parent_link) {
+		subsurface_create(view, subsurface);
+	}
+	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces_above, parent_link) {
 		subsurface_create(view, subsurface);
 	}
 
